src/models/data/data-set.ts

```diff
@@ -241,6 +248,56 @@ export const DataSet = types.model("DataSet", {
   function clearAllSelections() {
     self.attributeSelection.clear();
     self.caseSelection.clear();
+    self.cellSelection.clear();
+  }
+
+  function isCellSelected(cellIds: ICell) {
+    let cellSelected = false;
+    Array.from(self.cellSelection).forEach(cell => {
+      if (cell.attributeId === cellIds.attributeId && cell.caseId === cellIds.caseId) {
+        cellSelected = true;
+      }
+    });
+    return cellSelected;
+  }
+
+  function selectCases(caseIds: string[], select = true) {
+    if (select) self.attributeSelection.clear();
+    const ids: string[] = [];
+    caseIds.forEach(id => {
+      const pseudoCase = self.pseudoCaseMap[id];
+      if (pseudoCase) {
+        ids.push(...pseudoCase.childCaseIds);
+      } else {
+        ids.push(id);
+      }
+    });
+    if (select) {
+      self.attributeSelection.clear();
+      self.cellSelection.clear();
+    }
+    ids.forEach(id => {
+      if (select) {
+        self.caseSelection.add(id);
+      }
+      else {
+        self.caseSelection.delete(id);
+      }
+    });
+  }
+
+  function setSelectedCases(caseIds: string[]) {
+    clearAllSelections();
+    const ids: string[] = [];
+    caseIds.forEach(id => {
```

Comment:
This is essentially the same loop as lines 267-74 ... consider factoring out as a method?